src/components/Inventory/withdraw/flow/SelectPurposeForm.tsx

```diff
@@ -152,17 +120,23 @@ export default function SelectPurposeForm(props: SelectPurposeFormProps): JSX.El
   };
 
   const onNextHandler = async () => {
-    if (localRecords) {
-      const nurseryTransferInvalid = !validateNurseryTransfer();
-      const selectedNurseryInvalid = !validateSelectedNursery();
-      if (fieldsErrors.withdrawDate || nurseryTransferInvalid || selectedNurseryInvalid) {
-        return;
-      }
+    const nurseryTransferInvalid = !validateNurseryTransfer();
+    const selectedNurseryInvalid = !validateSelectedNursery();
+    if (fieldsErrors.withdrawDate || nurseryTransferInvalid || selectedNurseryInvalid) {
+      return;
     }
 
-    setRecords(localRecords.filter((rcd) => rcd.facilityId.toString() === selectedNursery));
-
-    onNext();
+    onNext({
```

Comment:
@constanzauanini looks like we got something wrong earlier.
The withdrawal is only one object, not an array. 
The batchWithdrawals property within the withdrawal object is the filtered down array of batches.
Fixed the code, it is actually a lot simpler now.